_rules/auto-update-text-efbfc7.md

```diff
@@ -0,0 +1,428 @@
+
+```html
+<body onload="start()">
+	<p>Random number: <span id="target">1</span></p>
+	<input type="button" id="control" onclick="pause()" value="Pause updates" />
+
+	<script type="text/javascript">
+		function change() {
+			var target = document.getElementById('target')
+			var number = Math.floor(Math.random() * 1000)
+			target.innerText = number
+		}
+
+		var updates
+		var updating = false
+		function start() {
+			var control = document.getElementById('control')
+			control.value = 'Pause updates'
+			updating = true
+			updates = setInterval(change, 3000)
+		}
+
+		function pause() {
+			if (updating) {
+				var control = document.getElementById('control')
+				control.value = 'Resume updates'
+				updating = false
+				clearInterval(updates)
+			} else {
+				start()
+			}
+		}
+	</script>
+</body>
+```
+
+#### Passed Example 3
+
+The text node (part of a progress bar) automatically updates every second after the page completes loading. Since the auto-updating can be considered essential because it takes place during a loading phase when there can be no interaction, a mechanism to stop the auto-updating is not required and the example passes.
+
+```html
+<html>
+	<head>
+		<title>Example page for rule efbfc7</title>
+		<style>
+			.progress-wrap,
+			.progress-bar {
+				height: 20px;
+				max-width: 300px;
+				position: relative;
+			}
+			.progress-wrap {
+				background: #eee;
+			}
+			.progress-bar {
+				background: #cce0ff;
+				width: 0;
+				transition: width 1s;
+			}
+			.progress-text {
+				position: absolute;
+				top: 0;
+				width: 100%;
+				text-align: center;
+			}
+			.choices {
+				margin: 3em;
+			}
+		</style>
+	</head>
+	<body onload="start()">
+		<p>
+			Wait for the bar to reach 100% then make your choice
+		</p>
+		<div class="progress-wrap">
+			<div class="progress-bar" id="pb-demo"></div>
+			<div class="progress-text" id="pb-text">0%</div>
+		</div>
+		<div class="choices">
+			<button type="button" id="packa" disabled>Pack A:</button>
+			<button type="button" id="packb" disabled>Pack B:</button>
+		</div>
+
+		<script type="text/javascript">
+			var n, updates
+
+			function updatePB() {
+				var bar = document.getElementById('pb-demo')
+				var text = document.getElementById('pb-text')
+				bar.style.width = n + '%'
+				text.innerHTML = n + '%'
+			}
+
+			function updateNum() {
+				if (n < 100) {
+					n = n + 5
+					updatePB(n)
+					if (n % 10 === 0) {
+						var buttonA = document.getElementById('packa')
+						var buttonB = document.getElementById('packb')
+						buttonA.innerHTML = 'Pack A: &euro;' + Math.floor(Math.random() * 1000)
+						buttonB.innerHTML = 'Pack B: &euro;' + Math.floor(Math.random() * 1000)
+					}
+				} else {
+					clearInterval(updates)
+					var buttonA = document.getElementById('packa')
+					var buttonB = document.getElementById('packb')
```

Comment:
Given that these variables are used in both branches of the main test, they should be moved to the top-level of the function instead.